### PR TITLE
Allow transaction_time to be undef

### PR DESCRIPTION
**Pull request by oschwald**

### CODE REVIEW

File: t/PaymentsAdvanced/Response/FromSilentPOST.t [hunk 1]
```diff
@@ -5,21 +5,40 @@ use Test::More;
 
 use Scalar::Util qw( blessed );
 use Test::Fatal;
+use WebService::PayPal::PaymentsAdvanced::Response::FromSilentPOST;
+();
```
Comment (horgh) [inline t/PaymentsAdvanced/Response/FromSilentPOST.t]: Should this be on the previous line?

File: lib/WebService/PayPal/PaymentsAdvanced/Role/HasTransactionTime.pm [hunk 2]
```diff
@@ -10,20 +10,22 @@ use feature qw( state );
...
     my $self = shift;
 
     state $time_zone
         = DateTime::TimeZone->new( name => 'America/Los_Angeles' );
-    my $dt = DateTime::Format::MySQL->parse_datetime(
-        $self->params->{TRANSTIME} );
+
+    return undef unless my $transtime = $self->params->{TRANSTIME};
+
+    my $dt = DateTime::Format::MySQL->parse_datetime($transtime);
```
Comment (horgh) [inline lib/WebService/PayPal/PaymentsAdvanced/Role/HasTransactionTime.pm:28]: MySQL? 😕

**Review by horgh — CHANGES_REQUESTED** [review]

File: t/PaymentsAdvanced/Response/FromSilentPOST.t (hunk 1 above)
Comment (oschwald) [reply t/PaymentsAdvanced/Response/FromSilentPOST.t]: Hmm, not sure how that happened. Thanks!

File: lib/WebService/PayPal/PaymentsAdvanced/Role/HasTransactionTime.pm (hunk 2 above)
Comment (oschwald) [reply lib/WebService/PayPal/PaymentsAdvanced/Role/HasTransactionTime.pm:28]: Yeah, I am not sure about that either, but I guess it works.

**Review by horgh — APPROVED** [review]

**Pull request merged**